org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java

```diff
@@ -1854,4 +1854,8 @@ public static boolean isGenerated(IMember member) {
 		return result;
 	}
 
+	public static boolean isUnnamedClass(IJavaElement element) {
+		return element.getElementType() == IJavaElement.TYPE && element.getElementName().startsWith("<unnamed_class$");

```

Comment:
`<unnamed_class$` seems very "implementation detail" specific. JDT really doesn't provide an API for that?

Reply:
I didn't add any method in the IJavaElement model to identify unnamed classes. The information should be available in the AST. In https://github.com/eclipse-jdtls/eclipse-jdt-core-incubator/pull/102 , I'll take a look into adding the information into the IJavaElement model. Maybe it's worth holding off until then to fix this?

Reply:
depends how fast it'll take you to get the fix in. Given the size of this patch, I'd be fine releasing with the current code, then improve it later with the proper API.

Reply:
Okay.